### PR TITLE
OpenThread upmerge July 5th

### PR DESCRIPTION
**Pull request by Damian-Nordic**
Regular upmerge to bring in recent infinite loop fix and the support for DNS service subtypes.

### CODE REVIEW

File: etc/cmake/options.cmake [hunk 1]
```diff
@@ -229,9 +229,14 @@ if(OT_LINK_RAW)
     target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_LINK_RAW_ENABLE=1")
 endif()
 
-option(OT_LINK_METRICS "enable link metrics")
-if (OT_LINK_METRICS)
-    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE=1")
+option(OT_LINK_METRICS_INITIATOR "enable link metrics initiator")
```
Comment (edmont) [inline etc/cmake/options.cmake:232]: Can you please take care of this change in the `zephyr/CMakeLists.txt` file?

**Review by edmont — CHANGES_REQUESTED** [review]
Thanks. We usually update the CMake changes for every upmerge. Changes need to be reflected in the Zephyr's manifest PR as well.

File: etc/cmake/options.cmake (hunk 1 above)
Comment (Damian-Nordic) [reply etc/cmake/options.cmake:232]: Good point. Added necessary changes.

**Review by edmont — APPROVED** [review]

**Review by rlubos — COMMENTED** [review]
Looks good, but could you please move changes in `zephyr/CMakeLists.txt` to a separate commit? It helps to keep track of the local changes in the repository.

**Comment by Damian-Nordic** [comment]
> Looks good, but could you please move changes in `zephyr/CMakeLists.txt` to a separate commit? It helps to keep track of the local changes in the repository.

Sure, I thought we don't want to have commits which are not "self-sufficient" and require fixes from subsequent changes, but I can split the change if it's better.

**Comment by rlubos** [comment]
> > Looks good, but could you please move changes in `zephyr/CMakeLists.txt` to a separate commit? It helps to keep track of the local changes in the repository.
> 
> Sure, I thought we don't want to have commits which are not "self-sufficient" and require fixes from subsequent changes, but I can split the change if it's better.

We don't use intermediate revisions in Zephyr, so it shouldn't affect bisectability.

**Comment by Damian-Nordic** [comment]
> We don't use intermediate revisions in Zephyr, so it shouldn't affect bisectability.

Done.

**Review by rlubos — APPROVED** [review]

**Pull request merged**